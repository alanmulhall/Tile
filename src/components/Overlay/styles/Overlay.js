import styled from 'styled-components';

export const OverlayStyled = styled.div`
  background: rgba(0, 0, 0, 0.65);
  box-sizing: border-box;
  color: white;
  left: 0;
  bottom: 0;
  padding: ${props => props.theme.spacing.xm} ${props => props.theme.spacing.m};
  position: absolute;
  height: ${props => props.fullscreen ? '100%' : 'auto'};
  width: 100%;
  z-index: ${props => props.theme.zIndex[0]};
`;
