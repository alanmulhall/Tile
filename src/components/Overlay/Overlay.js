import React from 'react';
import PropTypes from 'prop-types';
import { OverlayStyled } from './styles/Overlay';

const defaultProps = {
  children: null,
  className: null,
  fullscreen: true
};

const propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  fullscreen: PropTypes.bool
};

const Overlay = ({ children, className, fullscreen }) => (
  <OverlayStyled className={className} fullscreen={fullscreen}>
    {children}
  </OverlayStyled>
);

Overlay.defaultProps = defaultProps;
Overlay.propTypes = propTypes;

export default Overlay;
