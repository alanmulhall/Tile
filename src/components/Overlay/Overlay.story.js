import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, withKnobs } from '@storybook/addon-knobs';
import centered from '@storybook/addon-centered';
import withTests from '../../utils/withTests';
import Overlay from './Overlay';

const OverlayStory = () =>
  storiesOf('Overlay', module)
    .addDecorator(centered)
    .addDecorator(withKnobs)
    .addDecorator(withTests('Overlay'))
    .add('Default', () => <Overlay fullscreen={boolean('Fullscreen', true)} />);

export default OverlayStory;
