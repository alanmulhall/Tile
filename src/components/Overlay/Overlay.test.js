import React from 'react';
import 'jest-styled-components';
import renderer from '../../utils/rendererThemeProviderWrapper';
import shallow from '../../utils/shallowThemeProviderWrapper';
import Overlay from './Overlay';

describe('<Overlay />', () => {
  it('renders successfullly', () => {
    const tree = renderer(<Overlay />);
    expect(tree).toMatchSnapshot();
  });

  it('renders its children', () => {
    const wrapper = shallow(
      <Overlay>
        <p>Solo</p>
      </Overlay>
    );
    expect(wrapper.find('Overlay').children().length).toEqual(1);
  });

  describe('fullscreen prop is true', () => {
    it('fullscreen prop is true by default', () => {
      const wrapper = shallow(<Overlay />);
      expect(wrapper.find('Overlay').props().fullscreen).toEqual(true);
    });

    it('Text color is white', () => {
      const tree = renderer(<Overlay />);
      expect(tree).toHaveStyleRule('color', 'white');
    });

    it('has transparent background color', () => {
      const tree = renderer(<Overlay />);
      expect(tree).toHaveStyleRule('background', 'rgba(0,0,0,0.65)');
    });

    it('height is 100%', () => {
      const tree = renderer(<Overlay />);
      expect(tree).toHaveStyleRule('height', '100%');
    });
  });

  describe('fullscreen prop is false', () => {
    it('height is auto', () => {
      const tree = renderer(<Overlay fullscreen={false} />);
      expect(tree).toHaveStyleRule('height', 'auto');
    });
  });
});
