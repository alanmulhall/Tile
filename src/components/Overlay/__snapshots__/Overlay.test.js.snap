// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overlay /> renders successfullly 1`] = `
.c0 {
  background: rgba(0,0,0,0.65);
  box-sizing: border-box;
  color: white;
  left: 0;
  bottom: 0;
  padding: 0.6875rem 0.8125rem;
  position: absolute;
  height: 100%;
  width: 100%;
  z-index: 100;
}

<div
  className="c0"
/>
`;
